atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/TransactionLocksManager.java

```diff
@@ -16,31 +16,41 @@
 
 package com.palantir.atlasdb.transaction.impl;
 
+import com.google.common.collect.ImmutableSet;
 import com.google.common.collect.Sets;
+import com.palantir.atlasdb.transaction.impl.precommit.LockUnlocker;
 import com.palantir.atlasdb.transaction.impl.precommit.LockValidityChecker;
 import com.palantir.lock.v2.LockToken;
-import com.palantir.logsafe.Preconditions;
-import com.palantir.logsafe.SafeArg;
-import java.util.HashSet;
+import java.io.Closeable;
+import java.util.Collections;
+import java.util.LinkedHashSet;
 import java.util.Optional;
 import java.util.Set;
 import org.immutables.value.Value;
 
-public final class ImmutableTimestampLockManager {
+public final class TransactionLocksManager implements Closeable {
```

Comment:
I strongly feel like this should be a class that similarly locked as soon as we enter commitWrites or similar. As it's written, it's just weird to reason through locks being added willy nilly at any point. Given that we're not exposing this to end users yet, I think this is fine, and we should tackle this via some sort of CloseManager and proxies possibly

Reply:
Package private?